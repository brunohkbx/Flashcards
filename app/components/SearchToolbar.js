import * as React from 'react';
import PropTypes from 'prop-types';
import { Colors, Searchbar, Toolbar } from 'react-native-paper';

const SearchToolbar = ({ handleChangeText, value, handleIconPress }) => {
  return (
    <Toolbar style={{backgroundColor: Colors.white}}>
      <Searchbar
        placeholder="Search Decks"
        onChangeText={handleChangeText}
        value={value}
        icon="arrow-back"
        style={{elevation: 0}}
        onIconPress={handleIconPress}
        autoFocus
      />
    </Toolbar>
  );
};

SearchToolbar.propTypes = {
  value: PropTypes.string.isRequired,
  handleIconPress: PropTypes.func.isRequired
};

SearchToolbar.defaultProps = {
  value: ''
};

export default SearchToolbar;
