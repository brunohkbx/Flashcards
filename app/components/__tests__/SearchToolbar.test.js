import React from 'react';
import { shallow } from 'enzyme';
import SearchToolbar from '../SearchToolbar';

describe('SearchToolbar', () => {
  const setup = propOverrides => {
    const defaultProps = Object.assign({
      handleIconPress: jest.fn()
    }, propOverrides)

    const wrapper = shallow(<SearchToolbar {...defaultProps} />);

    const changeText = () => {
      wrapper
        .find('withTheme(Searchbar)')
        .simulate('changeText');
    }

    const iconPress = () => {
      wrapper
        .find('withTheme(Searchbar)')
        .simulate('iconPress');
    }

    return {
      wrapper,
      changeText,
      iconPress
    }
  }

  it('renders property', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  test('onChangeText calls handleChangeText', () => {
    const mockHandleChangeText = jest.fn();
    const { changeText } = setup({ handleChangeText: mockHandleChangeText });

    changeText();

    expect(mockHandleChangeText).toHaveBeenCalled();
  });

  test('onIconPress calls handleIconPress', () => {
    const mockHandleIconPress = jest.fn();
    const { iconPress } = setup({ handleIconPress: mockHandleIconPress });

    iconPress();

    expect(mockHandleIconPress).toHaveBeenCalled();
  });
});
