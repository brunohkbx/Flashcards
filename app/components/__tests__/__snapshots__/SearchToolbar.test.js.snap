// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchToolbar renders property 1`] = `
<withTheme(Toolbar)
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <withTheme(Searchbar)
    autoFocus={true}
    icon="arrow-back"
    onIconPress={[MockFunction]}
    placeholder="Search Decks"
    style={
      Object {
        "elevation": 0,
      }
    }
    value=""
  />
</withTheme(Toolbar)>
`;
